### PR TITLE
docs: adjust plugin dev guide example to write modified contents, fixes #3655

### LINKED ISSUE #3655: Plugin Development guide example doesn't update file content
### Version
3.5.1

### Environment info
```

Environment Info:

  System:
    OS: macOS 10.14.3
    CPU: (4) x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
  Binaries:
    Node: 8.15.0 - ~/.nvm/versions/node/v8.15.0/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v8.15.0/bin/npm
  Browsers:
    Chrome: 72.0.3626.121
    Firefox: 64.0
    Safari: 12.0.3
  npmPackages:
    @optoro/vue-cli-plugin-optics: * => 0.0.18 
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0-beta.2 
    @vue/babel-plugin-transform-vue-jsx:  1.0.0-beta.2 
    @vue/babel-preset-app:  3.5.1 
    @vue/babel-preset-jsx:  1.0.0-beta.2 
    @vue/babel-sugar-functional-vue:  1.0.0-beta.2 
    @vue/babel-sugar-inject-h:  1.0.0-beta.2 
    @vue/babel-sugar-v-model:  1.0.0-beta.2 
    @vue/babel-sugar-v-on:  1.0.0-beta.2 
    @vue/cli-overlay:  3.5.1 
    @vue/cli-plugin-babel: ^3.5.0 => 3.5.1 
    @vue/cli-plugin-eslint: ^3.5.0 => 3.5.1 
    @vue/cli-plugin-unit-jest: ^3.5.0 => 3.5.1 
    @vue/cli-service: ^3.5.0 => 3.5.1 
    @vue/cli-shared-utils:  3.5.1 
    @vue/component-compiler-utils:  2.6.0 
    @vue/eslint-config-standard: ^4.0.0 => 4.0.0 
    @vue/preload-webpack-plugin:  1.1.0 
    @vue/test-utils: 1.0.0-beta.29 => 1.0.0-beta.29 
    @vue/web-component-wrapper:  1.2.0 
    eslint-plugin-vue: ^5.0.0 => 5.2.2 
    jest-serializer-vue:  2.0.2 
    vue: ^2.6.6 => 2.6.9 
    vue-eslint-parser:  2.0.3 
    vue-hot-reload-api:  2.3.3 
    vue-jest:  3.0.4 
    vue-loader:  15.7.0 
    vue-router: ^3.0.1 => 3.0.2 
    vue-style-loader:  4.1.2 
    vue-template-compiler: ^2.5.21 => 2.6.9 
    vue-template-es2015-compiler:  1.9.1 
    vuex: ^3.0.1 => 3.1.0 
  npmGlobalPackages:
    @vue/cli: 3.5.1


```

### Steps to reproduce
Render the docs

### What is expected?
Correct information

### What is actually happening?
Incorrect information

---
This is just an issue with the Plugin Development guide docs.

https://github.com/vuejs/vue-cli/blob/dev/docs/dev-guide/plugin-dev.md#changing-main-file

Notice the example for `generator/index.js` writes the same content that is read from the file instead of writing the modified contents:

```js{10}
// generator/index.js

api.onCreateComplete(() => {
  const fs = require('fs')
  const contentMain = fs.readFileSync(api.entryFile, { encoding: 'utf-8' })
  const lines = contentMain.split(/\r?\n/g)

  const renderIndex = lines.findIndex(line => line.match(/render/))
  lines[renderIndex]  = `\n  router,`

  fs.writeFileSync(api.entryFile, contentMain, { encoding: 'utf-8' }) // XXX NO! This should render the modified `lines`
})
```

### PR DESCRIPTION
**Pull request by akrawchyk**
fixes #3655

### CODE REVIEW

**Comment by NataliaTepluhina** [comment]
@akrawchyk thank you so much for spotting this! I think I missed
```js
    contentMain = lines.join('\n');
```
In my example but including end of line makes it even better.

**Comment by akrawchyk** [comment]
@NataliaTepluhina no prob 😄 , I can update the example to use either `EOL` or `\n` everywhere, up to you.

**Comment by NataliaTepluhina** [comment]
I think EOL is more consistent in this case so let's leave it your way :wink: 

@sodatea please have a final look :bowing_woman:

**Comment by sodatea** [comment]
The best option is to maintain the line-ending (like prettier https://github.com/prettier/prettier/pull/472)
But that may be too much for an example snippet so I think EOL is good enough.

(sidenote: Airbnb Style Guide enforces `LF` as line ending, while create-react-app uses `os.EOL` everywhere https://github.com/facebook/create-react-app/pull/6610, other common style guides has no strong opinion on this, which means most people may just use os default)

File: docs/dev-guide/plugin-dev.md [hunk 1]
```diff
@@ -234,18 +234,19 @@ api.onCreateComplete(() => {
...
 // generator/index.js
 
 api.onCreateComplete(() => {
+  const { EOL } = require('os')
   const fs = require('fs')
   const contentMain = fs.readFileSync(api.entryFile, { encoding: 'utf-8' })
   const lines = contentMain.split(/\r?\n/g)
 
   const renderIndex = lines.findIndex(line => line.match(/render/))
   lines[renderIndex] += `\n  router,`
```
Comment (sodatea) [inline docs/dev-guide/plugin-dev.md]: Now that we join the lines with `os.EOL`, this string concatenation has to be changed too:
```suggestion
   lines[renderIndex] += `${EOL}  router,`
```
or
```suggestion
   lines.splice(renderIndex, 0, `  router,`)
```

**Pull request merged**